File: python/src/sort/cocktail_sort.py


def cocktail_sort(in_array):
    i, k = 0, len(in_array)-1
    while i < k:
        for j in range(i,k+1):
            if in_array[j] < in_array[i]:
                in_array[i], in_array[j] = in_array[j], in_array[i]
        for m in range(k, i, -1):
            if in_array[m] > in_array[k]:
                in_array[m], in_array[k] = in_array[k], in_array[m]
        i += 1
        k -= 1
    return in_array

File: c#/Sort/CocktailSort.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Sort
{
    public class CocktailSort : ISort
    {

        public ArrayList Sort(ArrayList inArray)
        {
            int i = 0, k = inArray.Count - 1;
            while(i < k)
            {
                for(int j=i,l=k+1; j<l; j++)
                {
                    if ((int)inArray[j] < (int)inArray[i])
                        Swap(i, j, ref inArray);
                }
                for(int m=k; m>i; m--)
                {
                    if( (int)inArray[m] > (int)inArray[k] )
                        Swap(m, k, ref inArray);
                }

                i++;
                k--;
            }

            return inArray;
        }

        public void Swap(int i, int j, ref ArrayList inArray)
        {
            var tmp = inArray[j];
            inArray[j] = inArray[i];
            inArray[i] = tmp;
        }
    }
}
